### PR TITLE
Features: #123 Array like options

### PR DESCRIPTION
**Pull request by stephane-segning**

### CODE REVIEW

**Comment by adrien2p** [comment]
I ll do it as soon as I can 😊 thanks a lot. We will also have to update the documentation to mention that you can now pass an array of object/function. I think ill also do some helpers for the test for example the container mock and things like that

**Comment by stephane-segning** [comment]
Good ideas!
- Implementing new examples would be much more helpful for newbies.
- Having better doc is a foundation for us but for other developers too for having good code. Tell me when you're doing, I'll like to help :)

**Comment by adrien2p** [comment]
If you have any idea to make the doc better i would be glad to look at your suggestions 🎉very happy to have you help me on the plugin. As you know i am pretty busy and i am trying to find time to look at your current prs 🤣

File: packages/medusa-plugin-auth/src/api/index.ts [hunk 1]
```diff
@@ -9,23 +9,43 @@ import FirebaseStrategy from '../auth-strategies/firebase';
...
-	routers.push(...FacebookStrategy.getRouter(configModule, options));
-	routers.push(...LinkedinStrategy.getRouter(configModule, options));
-	routers.push(...FirebaseStrategy.getRouter(configModule, options));
-	routers.push(...Auth0Strategy.getRouter(configModule, options));
-	routers.push(...AzureStrategy.getRouter(configModule, options));
+	for (const opt of options) {
+		const option = await handleOption(opt, configModule);
+
+		switch (option.type) {
+			case 'azure_oidc':
```
Comment (adrien2p) [inline packages/medusa-plugin-auth/src/api/index.ts:26]: suggestion: I would prefer to move the type to an enum wdyt?

File: packages/medusa-plugin-auth/src/loaders/index.ts [hunk 2]
```diff
@@ -9,14 +9,44 @@ import FireaseStrategy from '../auth-strategies/firebase';
...
+	authOptions: AuthOptionsWrapper | AuthOptionsWrapper[],
+) {
 	const configModule = container.resolve('configModule') as ConfigModule;
+	if (Array.isArray(authOptions)) {
+		for (const opt of authOptions) {
+			await handleStrategyLoading(opt, configModule, container);
+		}
+	} else {
+		await handleStrategyLoading(authOptions, configModule, container);
+	}
```
Comment (adrien2p) [inline packages/medusa-plugin-auth/src/loaders/index.ts:23]:
```suggestion
    const authOptions_ = Array.isArray(authOptions) ? authOptions : [authOptions]
	for (const opt of authOptions) {
	  await handleStrategyLoading(opt, configModule, container);
    }
```

**Review by adrien2p — COMMENTED** [review]
Could you run the lint script please to remove all lint issues and simplify the review 💪

**Comment by stephane-segning** [comment]
@adrien2p please review it again

**Comment by adrien2p** [comment]
Yes i will as soon as i can, i do it in my free time and i am a dad as well while i work at medusa 😉

**Review by adrien2p — APPROVED** [review]
LGTM, did have you been able to test with some strategies? for me it looks good, once I merge it we can focus on the doc part of it wdyt?

**Comment by stephane-segning** [comment]
Ive not tested it yet through and through. Still doing it though, but I'm not having errors till now. So I believe it's ok. 

Also working on a doc directly after is a good idea. Don't just deploy it for now 😊

**Pull request merged**